SpatialGDK/Source/SpatialGDKTests/SpatialGDK/Interop/Connection/SpatialWorkerConnectionTest.cpp

```diff
@@ -0,0 +1,370 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Tests/TestDefinitions.h"
+
+#include "SpatialWorkerConnection.h"
+
+#include "Interop/SpatialOutputDevice.h"
+
+#include "SpatialGDKServices/LocalDeploymentManager/LocalDeploymentManagerUtilities.h"
+
+#include "CoreMinimal.h"
+
+#define WORKERCONNECTION_TEST(TestName) \
+	GDK_TEST(Core, SpatialWorkerConnection, TestName)
+
+using namespace SpatialGDK;
+
+namespace
+{
+bool bClientConnectionProcessed = false;
+bool bServerConnectionProcessed = false;
+double MAX_WAIT_TIME = 10.0;
+
+void ConnectionProcessed(bool bConnectAsClient)
+{
+	if (bConnectAsClient)
+	{
+		bClientConnectionProcessed = true;
+	}
+	else
+	{
+		bServerConnectionProcessed = true;
+	}
+}
+
+void StartSetupConnectionConfigFromURL(USpatialWorkerConnection* Connection, const FURL& URL, bool& bOutUseReceptionist)
+{
+	bOutUseReceptionist = !(URL.Host == SpatialConstants::LOCATOR_HOST || URL.HasOption(TEXT("locator")));
+	if (bOutUseReceptionist)
+	{
+		Connection->ReceptionistConfig.SetReceptionistHost(URL.Host);
+	}
+	else
+	{
+		FLocatorConfig& LocatorConfig = Connection->LocatorConfig;
+		LocatorConfig.PlayerIdentityToken = URL.GetOption(*SpatialConstants::URL_PLAYER_IDENTITY_OPTION, TEXT(""));
+		LocatorConfig.LoginToken = URL.GetOption(*SpatialConstants::URL_LOGIN_OPTION, TEXT(""));
+	}
+}
+
+void FinishSetupConnectionConfig(USpatialWorkerConnection* Connection, const FString& WorkerType, const FURL& URL, bool bUseReceptionist)
+{
+	// Finish setup for the config objects regardless of loading from command line or URL
+	if (bUseReceptionist)
+	{
+		// Use Receptionist
+		Connection->SetConnectionType(ESpatialConnectionType::Receptionist);
+
+		FReceptionistConfig& ReceptionistConfig = Connection->ReceptionistConfig;
+		ReceptionistConfig.WorkerType = WorkerType;
+
+		const TCHAR* UseExternalIpForBridge = TEXT("useExternalIpForBridge");
+		if (URL.HasOption(UseExternalIpForBridge))
+		{
+			FString UseExternalIpOption = URL.GetOption(UseExternalIpForBridge, TEXT(""));
+			ReceptionistConfig.UseExternalIp = !UseExternalIpOption.Equals(TEXT("false"), ESearchCase::IgnoreCase);
+		}
+	}
+	else
+	{
+		// Use Locator
+		Connection->SetConnectionType(ESpatialConnectionType::Locator);
+		FLocatorConfig& LocatorConfig = Connection->LocatorConfig;
+		FParse::Value(FCommandLine::Get(), TEXT("locatorHost"), LocatorConfig.LocatorHost);
+		LocatorConfig.WorkerType = WorkerType;
+	}
+}
+}
```

Comment:
```suggestion
} // anonymous namespace
```
Nit: not sure what our policy on this was

Reply:
Added.